pkg/docker/docker.go

```diff
@@ -106,26 +120,25 @@ type Docker interface {
 // Client contains all methods called on the go Docker
 // client.
 type Client interface {
-	RemoveImage(name string) error
-	InspectImage(name string) (*docker.Image, error)
-	PullImage(opts docker.PullImageOptions, auth docker.AuthConfiguration) error
-	CreateContainer(opts docker.CreateContainerOptions) (*docker.Container, error)
-	AttachToContainerNonBlocking(opts docker.AttachToContainerOptions) (docker.CloseWaiter, error)
-	StartContainer(id string, hostConfig *docker.HostConfig) error
-	WaitContainer(id string) (int, error)
-	UploadToContainer(id string, opts docker.UploadToContainerOptions) error
-	DownloadFromContainer(id string, opts docker.DownloadFromContainerOptions) error
-	RemoveContainer(opts docker.RemoveContainerOptions) error
-	CommitContainer(opts docker.CommitContainerOptions) (*docker.Image, error)
-	CopyFromContainer(opts docker.CopyFromContainerOptions) error
-	BuildImage(opts docker.BuildImageOptions) error
-	InspectContainer(id string) (*docker.Container, error)
-	Ping() error
+	ContainerWait(ctx context.Context, containerID string) (int, error)
+	// no CommitContainer in kube_docker_client.go, but engine-api has ContainerCommit
+	ContainerCommit(ctx context.Context, container string, options dockertypes.ContainerCommitOptions) (dockertypes.ContainerCommitResponse, error)
+	// CopyToContainer is the UploadToContainer equivalent
+	CopyToContainer(ctx context.Context, container, path string, content io.Reader, opts dockertypes.CopyToContainerOptions) error
+	// no CopyFromContainer in kube_docker_client.go ... exists in engine-api, is the DownloadFromContainer equivalent
+	CopyFromContainer(ctx context.Context, container, srcPath string) (io.ReadCloser, dockertypes.ContainerPathStat, error)
+	// no BuildImage in kube_docker_client.go
+	ImageBuild(ctx context.Context, buildContext io.Reader, options dockertypes.ImageBuildOptions) (dockertypes.ImageBuildResponse, error)
+	// no Ping in kube_docker_client.go or engine-api ... going to borrow a small bit from fsouza
```

Comment:
There *used* to be a Ping method in the Client interface.  But yes, overall, these comments were work in progress type reminders for me in the end.  I agree they can be removed.  Doing that for all of them.